### ISSUE TITLE
Enhancement: Allow the HitNoteScriptEvent/NoteScriptEvent to recieve/get the character who hit the note

### ISSUE THREAD

**Issue by Starexify** [FunkinCrew/Funkin#3921]
### Issue Checklist

- [X] I have properly named my enhancement
- [X] I have checked the Issues/Discussions pages to see if my enhancement has already been suggested

### What is your suggestion, and why should it be implemented?

I was trying to somehow get the character who hit the NoteKind with onNoteHit function, but I don't seem to find a way to get the id/character who hit the note. I am not sure if there is any way to make that possible with only onNoteHit event.

Reason for this implementation would be for easier data recieving in events, so NoteKinds are based on the character who hit the note aswell.

If anyone has a way, would also appreciate 👍

Update: I found a way to do this, but I have no idea how optimal it is:
```haxe
    override function onNoteHit(event:HitNoteScriptEvent) {
        if (event.note.noteData.getMustHitNote()) //   Returns true when if the note is on player side
    }
```

**Comment by EliteMasterEric** [comment]
The provided snippet is the correct/optimal way to determine which character hit the note. You can see that snippet used in various scripts used by the base game.

**Comment by Starexify** [comment]
> The provided snippet is the correct/optimal way to determine which character hit the note. You can see that snippet used in various scripts used by the base game. Closing as resolved.

Why not changing the name better into: `isPlayerNote` or something similar because that is kinda ambiguous ?

**Comment by AbnormalPoof** [comment]
That would break existing mods that use it.

**Comment by Starexify** [comment]
> That would break existing mods that use it.

Well if it gets documented (changelog) then what is the problem with that 🤔

**Comment by EliteMasterEric** [comment]
I agree the name is a bit ambiguous (since it doesn't take into account mines, etc) but the value is too widely used to be renamed without sweeping and unnecessary breaking changes.
